x-pack/platform/plugins/shared/lens/public/app_plugin/csv_download_provider/csv_download_provider.tsx

```diff
@@ -109,12 +109,11 @@ export const downloadCsvLensShareProvider = ({
   uiSettings,
   formatFactoryFn,
   atLeastGold,
-}: DownloadPanelShareOpts): ExportShare => {
+}: DownloadPanelShareOpts): RegisterShareIntegrationArgs<ExportShare> => {
   return {
-    shareType: 'integration',
     id: 'csvDownloadLens',
     groupId: 'export',
-    config({ sharingData }) {
+    getShareIntegrationConfig: async ({ sharingData }) => {
       // TODO fix sharingData types
```

Comment:
There's an opportunity here to move the function body into a different file, so it's only ever loaded when we are certain the current user is eligible to use the CSV share integration and the action button has been clicked.

Reply:
The CSV Lens export works in a slightly different way that the regular CSV reporting system and there's no current permission setting to disable it yet.
Consider also that the CSV data is already on the client, so "disable" in this case would only mean "hide the button and make it a bit harder to download what is already in inspector".